apiary-metastore-listener/src/test/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListenerTest.java

```diff
@@ -237,12 +288,18 @@ public void onAlterTable() throws MetaException {
 
     assertThat(publishRequest.getMessage(), is("{\"protocolVersion\":\""
         + PROTOCOL_VERSION
-        + "\",\"eventType\":\"ALTER_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"new_some_table\",\"oldTableName\":\"some_table\"}"));
+        + "\",\"eventType\":\"ALTER_TABLE\",\"dbName\":\"some_db\",\"tableName\":\"new_some_table\",\"tableLocation\":\"s3://table_location_1\",\"hkaasParameters\":\"{HKAAS_EXPIRY_DAYS=5, HKAAS_ENABLED=true}\",\"oldTableName\":\"some_table\",\"oldTableLocation\":\"s3://table_location\"}"));
   }
 
-  private StorageDescriptor createStorageDescriptor(List<FieldSchema> partitionKeys) {
-    return new StorageDescriptor(partitionKeys, "s3://test_location", "ORC", "ORC", false, 2, new SerDeInfo(),
+  @Test
+  public void testEnvironmentVariablesAreBeingSet() {
```

Comment:
Why are we testing this? This is effectively testing the system rule and not our code right?

Reply:
I saw this `TODO: test for setting ARN via environment variable` and was wondering about the same but added this test anyway.

Can you please elaborate on what do you expect from that TODO ?

Reply:
Testing what happens in the listener code when this is set and isn't, not sure how easy this is since it's currently only used to create a `PublishRequest` object on line 200. Does this get and use the right value would be the first thing to test. Then what would happen if the value wasn't set? Would this throw a NPE? If so we should do some validation on it being set earlier and throw a more descriptive error etc. Probably out of scope of this PR but would be nice to clean this up at some point.

Reply:
Ok. we discussed it offline and decided to put the TODO again as its not straightforward to test it with current implementation.